includes/DataObjects/User.php

```diff
@@ -779,213 +561,15 @@ public function isCommunityUser()
-            $this->dbObject->prepare("UPDATE user SET oauthidentitycache = :identity WHERE id = :id;")
-                ->execute(array(":id" => $this->id, ":identity" => $this->oauthidentitycache));
-        }
-        catch (UnexpectedValueException $ex) {
-            $this->identityCache = null;
-            $this->oauthidentitycache = null;
-            $this->dbObject->prepare("UPDATE user SET oauthidentitycache = NULL WHERE id = :id;")
-                ->execute(array(":id" => $this->id));
-
-            SessionAlert::warning("OAuth error getting identity from MediaWiki: " . $ex->getMessage());
-        }
-    }
-
-    /**
-     * @return bool
-     * @todo move me to a collaborator
-     */
-    public function oauthCanUse()
-    {
-        try {
-            return in_array('useoauth', $this->getOAuthIdentity()->grants);
-        }
-        catch (Exception $ex) {
-            return false;
-        }
-    }
-
-    /**
-     * @return bool
-     * @todo move me to a collaborator
-     */
-    public function oauthCanEdit()
-    {
-        try {
-            return in_array('useoauth', $this->getOAuthIdentity()->grants)
-            && in_array('createeditmovepage', $this->getOAuthIdentity()->grants)
-            && in_array('createtalk', $this->getOAuthIdentity()->rights)
-            && in_array('edit', $this->getOAuthIdentity()->rights)
-            && in_array('writeapi', $this->getOAuthIdentity()->rights);
-        }
-        catch (Exception $ex) {
-            return false;
-        }
-    }
-
-    /**
-     * @return bool
-     * @todo move me to a collaborator
-     */
-    public function oauthCanCreateAccount()
-    {
-        try {
-            return in_array('useoauth', $this->getOAuthIdentity()->grants)
-            && in_array('createaccount', $this->getOAuthIdentity()->grants)
-            && in_array('createaccount', $this->getOAuthIdentity()->rights)
-            && in_array('writeapi', $this->getOAuthIdentity()->rights);
-        }
-        catch (Exception $ex) {
-            return false;
-        }
-    }
-
-    /**
-     * @return bool
-     * @todo     move me to a collaborator
-     * @category Security-Critical
-     */
-    protected function oauthCanCheckUser()
-    {
-        if (!$this->isOAuthLinked()) {
-            return false;
-        }
-
-        try {
-            $identity = $this->getOAuthIdentity();
-
-            return in_array('checkuser', $identity->rights);
-        }
-        catch (Exception $ex) {
-            return false;
-        }
-    }
-
-    #endregion
-
     /**
      * Gets a hash of data for the user to reset their password with.
      * @category Security-Critical
      * @return string
      */
     public function getForgottenPasswordHash()
     {
-        return md5($this->username . $this->email . $this->welcome_template . $this->id . $this->password);
+        // FIXME
+        return md5($this->username . $this->email . $this->welcome_template . $this->id);
```

Comment:
I am fully aware that this is too weak. I'm not planning on fixing it in this review, but I've raised #528 to fix it prior to go-live.